statshouse.hpp

```diff
@@ -695,6 +708,473 @@ class TransportUDP : public TransportUDPBase {
+				auto &ptr = queue.front();
+				{
+					std::lock_guard<std::mutex> bucket_lock{ptr->mu};
+					if (!ptr->value.empty()) {
+						buffer[count] = alloc_bucket(ptr->key, ptr->timestamp);
+						std::swap(buffer[count]->value, ptr->value);
+						++count;
+					}
+				}
+				// pop front
+				if (ptr.use_count() > 2) {
+					// there are clients holding bucket reference, move bucket back
+					ptr->timestamp = time_now();
+					queue.push_back(std::move(ptr));
+					ptr = queue.back();
+					ptr->queue_ptr = &queue.back();
+				} else {
+					// removing bucket with non-zero "queue_ptr" implies removal from dictionary
+					if (ptr->queue_ptr) {
+						ptr->queue_ptr = nullptr;
+						buckets.erase(string_view{ptr->key.buffer, ptr->key.buffer_pos});
+					}
+					// unused bucket goes into freelist
+					freelist.emplace_back(std::move(ptr));
+				}
+				queue.pop_front();
+			}
+		}
+		for (size_t i = 0; i < count; ++i) {
+			auto &ptr = buffer[i];
+			{
+				std::lock_guard<std::mutex> transport_lock{transport_mu};
+				auto &v = ptr->value;
+				if (!v.values.empty()) {
+					ptr->key.write_values(v.values.data(), v.values.size(), v.count, timestamp);
+				} else if (!v.unique.empty()) {
+					ptr->key.write_unique(v.unique.data(), v.unique.size(), v.count, timestamp);
+				} else if (v.count) {
+					ptr->key.write_count(v.count, timestamp);
+				}
+				v.count = 0;
+				v.values.clear();
+				v.unique.clear();
+			}
+			std::lock_guard<std::mutex> lock{mu};
+			freelist.emplace_back(std::move(ptr));
+		}
+		return count == size;
+	}
+	std::shared_ptr<bucket> get_or_create_bucket(const TransportUDP::MetricBuilder &key, uint32_t timestamp = 0) {
+		std::lock_guard<std::mutex> lock{mu};
+		auto it = buckets.find(string_view{key.buffer, key.buffer_pos});
+		if (it != buckets.end()) {
+			return it->second;
+		}
+		auto ptr = alloc_bucket(key, timestamp ? timestamp : time_now());
+		queue.emplace_back(buckets.emplace(string_view{ptr->key.buffer, ptr->key.buffer_pos}, ptr).first->second);
+		ptr->queue_ptr = &queue.back();
+		return ptr;
+	}
+	std::shared_ptr<bucket> alloc_bucket(const TransportUDP::MetricBuilder &key, uint32_t timestamp) {
+		if (freelist.empty()) {
+			return std::make_shared<bucket>(key, timestamp);
+		}
+		auto ptr = std::move(freelist.back());
+		std::memcpy(&ptr->key, &key, sizeof(TransportUDP::MetricBuilder));
+		ptr->timestamp = timestamp;
+		freelist.pop_back();
+		return ptr;
+	}
+	uint32_t time_now() const {
+		uint32_t t{time_external};
+		return t != 0 ? t : static_cast<uint32_t>(std::time(nullptr));
+	}
+	struct string_view_hash {
+		size_t operator()(string_view a) const {
+			return hash_bytes(a.data(), a.size(), 0);
+		}
+	};
+	struct string_view_equal {
+		bool operator()(string_view a, string_view b) const {
+			return a.size() == b.size() &&
+				   !std::memcmp(a.data(), b.data(), a.size());
+		}
+	};
```

Comment:
`std::string_view` already has `std::hash` specialization and equality operator defined. Please, move this code to `statshouse::string_view` stub under an `#ifdef`

Reply:
Yes, it would be better to use standart hash, but 
1. we target C++11 and
2. wyhash is 2x times faster